src/utils.ts

```diff
@@ -25,17 +26,19 @@ const AsyncFunctionConstructor = (async () => {}).constructor;
 /**
  * an async function check method only consider runtime support async syntax
  */
-export const isAsyncFunction = async (task: Task) => {
-  if (task.fn.constructor === AsyncFunctionConstructor) {
+export const isAsyncFunction = (fn: Fn) => fn.constructor === AsyncFunctionConstructor;
+
+export const isAsyncTask = async (task: Task) => {
+  if (isAsyncFunction(task.fn)) {
     return true;
   }
   try {
     if (task.opts.beforeEach != null) {
       await task.opts.beforeEach.call(task);
     }
     const call = task.fn();
-    const result = call instanceof Promise;
-    if (result) {
+    const result = typeof call?.then === 'function';
+    if (result && call?.catch) {
```

Comment:
You could actually send it to nirvana by doing

call.then(() => {}, () => {})

So you can save your check for catch. E.g. some fastify classes are thenable but have no .catch.

Reply:
Then why do you check for catch?

Reply:
Some tests throw unhandled errors